ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala

```diff
@@ -298,13 +307,13 @@ final class SandboxServer(
                 defaultLedgerConfiguration = defaultConfiguration,
                 commandConfig = config.commandConfig,
                 submissionConfig = config.submissionConfig,
-                optTimeServiceBackend = timeServiceBackendO
-                  .map(TimeServiceBackend.withObserver(_, indexAndWriteService.publishHeartbeat)),
+                optTimeServiceBackend = observingTimeServiceBackend,
                 metrics = metrics,
                 healthChecks = healthChecks,
                 seedService = seedService,
               )(mat, esf, logCtx)
               .map(_.withServices(List(resetService(ledgerId, authorizer, executionContext)))),
+          // NOTE: Re-use the same port after reset.
```

Comment:
What do you mean by this comment? The code says if you already have a port set then don't care about the port specified in the config. If you would like to imply that `currentPort` is the port saved before reset then please rename the parameter accordingly.

Reply:
This comment was just moved from the code above because it no longer fit; I didn't change it, and I'd rather not change it in this PR.

Reply:
OK -- I'm just pointing out this comment is rather confusing than helping.